wiki/src/main/resources/XWiki/PDFViewerMacro.xml

```diff
@@ -187,10 +194,8 @@
     #set ($authorRightsNeeded = false)
     #if ("$!docname" != '')
       #set ($document = $xwiki.getDocument($docname))
-      #if (!$document.getAttachment($file) &amp;&amp; $asAuthor &amp;&amp; $xwiki.hasProgrammingRights() &amp;&amp;
-          $xwiki.exists('XWiki.PDFViewerService'))
-        #set ($document = $xwiki.getDocumentAsAuthor($docname))
-        #set ($authorRightsNeeded = true)
+      #if (!$document.getAttachment($file))
```

Comment:
The problem with this condition is that it can be true when the document is accessible (viewable) but the attachment simply doesn't exist (was deleted in the mean time or simply didn't exist because the user typed the wrong file name). The question is whether it's fine / not misleading to try to load the document as author in this case (i.e. no side effects). I feel it would be cleaner / simpler to either test only the return document (knowing that ``getDocument()`` returns ``null`` when the document is not accessble) or to test explicitly if the current user has view right on the current document (using the security script service).